### DISCUSSION TITLE
regular user login?

### DISCUSSION THREAD

**Discussion by ricardogsilva** [fief-dev/fief#128] [Q&A] [answered]
I've set up a self-hosted fief instance locally, using docker-compose.

Afterwards I:

- Created a new fief workspace named `test`. From reading the discussion in #16 I understood it would be best if I left the main `fief` workspace only for the fief admin user, so I created a workspace for my tests.
- Created a new user `user1@example.com` in this `test` workspace;
- Tried out the _API Example_ section of the official fastapi integration tutorial shown at https://docs.fief.dev/integrate/python/fastapi/ - it works quite well. This left me very interested in using fief :smile_cat: 
- Using my admin user, I added some custom user fields and set some default values for the previsouly created `user1` user.

Now I'm trying to login to my `test` fief workspace as this `user1` in order to see how I can set/change the value of these custom user fields as a regular user. I can't seem to do this though, since I'm not even being shown a login page.

Whenever trying to visit:

http://test.localhost:8000/

I get back an `HTTP 404` status code and I'm greeted with the following JSON response in my browser

```json
{
  "detail": "Not Found"
}
```

This seems odd to me, as I'd expect this to be the main page of my fief workspace. Moreover, I would expect to be able to visit this page with my browser and be offered some HTML with links to other relevant pages, like a login page.

In trying to guess how I would login to the fief workspace, I'm trying to visit:

http://test.localhost:8000/login

I get back `HTTP 400` status code, with an `Invalid login session` error page, like this:

![image](https://user-images.githubusercontent.com/732010/208887019-275d5d84-67ee-4617-b6db-3410f53c991a.png)

I would expect fief to allow regular users to log in and be able to edit their own properties, so I'm wondering if this is some bug, or if I missed something. Perhaps I'm trying the wrong URLs? But I haven't been able to find them in the documentation either.

**Comment by frankie567** [comment, marked as answer]
You did everything correctly, actually 🙂

However, for now, we don't have the ability to directly access the authentication pages from a given URL. You should always generate an authorization URL from one of our client SDK (using a valid Client ID and Secret on your test workspace).

Besides, there is currently no UI allowing the user to edit its profile. That's something you should do from the API, for example by using the [`update_profile`](https://fief-dev.github.io/fief-python/fief_client.html#Fief.update_profile) method of the SDK.

Those are limitations I have in mind, and I plan to implement a proper "user dashboard" users can access without an application context.

**Reply by ricardogsilva** [reply]
Cool, thanks for the quick reply. I'll manage my expectations then :smile: 
In the meanwhile, I guess I will implement user profile editing in the fastapi app then.
